src/main/java/io/bittiger/ads/activity/AdsSelect.java

```diff
@@ -0,0 +1,51 @@
+package io.bittiger.ads.activity;
+
+import io.bittiger.ads.model.Ad;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.lang.String;
+
+public class AdsSelect {
+    private static AdsSelect instance = null;
+
+    protected AdsSelect() {
+    }
+
+    public static AdsSelect getInstance() {
+        if (instance == null) {
+            instance = new AdsSelect();
+        }
+        return instance;
+    }
+
+    public List<Ad> getMatchedAds(String[] keywords) {
+        List<Ad> ads = new ArrayList<Ad>();
+        if (keywords == null || keywords.length == 0) {
+            return ads;
+        }
+
+        for (String keyword : keywords) {
+            ads.addAll(pullMatchedAdsFromMemcached(keyword));
+        }
+
+        return ads;
+    }
+
+    public List<Ad> pullMatchedAdsFromMemcached(String keyword) {
+        List<Ad> ads = new ArrayList<Ad>();
+        if (keyword == null || keyword.isEmpty()) {
+            return ads;
+        }
+
+        /* tmp ads*/
+        long key = 0;
+        for (char c : keyword.toCharArray()) {
+            key += c - 'a';
+        }
+        /* tmp ads*/
+        ads.add(AdsDao.getInstance().getAd(key));

```

Comment:
The method "pullMatchedAdsFromMemcached" has not been completed, since I am stuck with pull List<ad> from memcached.  Here is pseudo code to present some ads are added here.  In my opinion, <String keyword, List<Ad> ads> is stored in memchached, but we have not implemented it.  The correct code should be something like "List<ad> res = memcached.get(keyword)".  This is the point in my commit comments 3 which should be discussed tonight.